shark-hprof/src/main/java/shark/Hprof.kt

```diff
@@ -25,25 +32,80 @@ class Hprof private constructor(
   val fileLength: Long
 ) : Closeable {
 
+  private val randomAccessBuffer = Buffer()
+
+  /**
+   * When true, enables a more efficient way of reading records with random access.
+   *
+   * Default is false.
+   */
+  var readFullRecords: Boolean = false
+    set(setToRandomAccess) {
+      if (setToRandomAccess == field) {
+        return
+      }
+      require(randomAccessBuffer.size() == 0L)
+      if (setToRandomAccess) {
+        reader.source = randomAccessBuffer
+        reader.position = 0
+      } else {
+        reader.source = source
+        // This ensures moveReaderTo actually moves.
```

Comment:
This is confusing, why do we need to call this before invoking `moveReaderTo(reader.startPosition)` ?

Reply:
Because of this check in moveReaderTo:

```
      if (currentPosition == newPosition) {
        return
      }
```